core/shared/src/main/scala/scalaz/effect/KleisliIO.scala

```diff
@@ -190,9 +192,13 @@ object KleisliIO {
   }
 
   private[effect] final class Pure[E, A, B](apply0: A => IO[E, B]) extends KleisliIO[E, A, B] {
+    override val run: A => IO[E, B] = a => apply(a)
+
     override final def apply(a: A): IO[E, B] = apply0(a)
```

Comment:
Delete all the `def apply()` method on `Impure` and `Pure`, because it's redundant with `run`.

Reply:
So i also delete the def apply() from KleisliIO?

Reply:
Yes, that's right, it's no longer necessary since a user can just call `run` to achieve the same effect.